include/jemalloc/internal/extent.h

```diff
@@ -125,6 +125,9 @@ extent_can_acquire_neighbor(edata_t *edata, rtree_contents_t contents,
 			return false;
 		}
 	}
+	if (edata_guarded_get(edata) != edata_guarded_get(neighbor)) {
+		return false;
+	}
```

Comment:
I was thinking we would potentially try to acquire the head of the exponential mapping from san_bump, but that one is marked as `EXTENT_IS_HEAD`, so it would be filtered out by of the previous checks.

Will change this to an assertion